### ISSUE TITLE
Include example code of PolyEditTool and PolyDrawTool Docs 

### ISSUE THREAD

**Issue by scaine1** [bokeh/bokeh#9962]
Currently the page https://docs.bokeh.org/en/latest/docs/user_guide/tools.html shows gif images of how to use the tools (which are great!). 

However, they don't include the code on how to use these tools in practice.

It took me an embarrassingly long time work workout how to use the PolyEditTool.

It would be great if the docs also included sample code on how to use the tools as well as the gif image.

These examples already exist in ```bokeh/sphinx/source/docs/user_guide/examples/tools_poly_draw.py```
and 
```bokeh/sphinx/source/docs/user_guide/examples/tools_poly_edit.py```

Could the code in these examples be linked directly to  https://docs.bokeh.org/en/latest/docs/user_guide/tools.html

**Comment by bryevdv** [comment]
@scaine1 definitely, I would like to resolve this in a general way by updating the `bokeh-plot` sphinx directive we created to take an option to include the link to the source file automatically. The directive can already include the actual source text but we deemed those examples a bit long to include wholesale inline in the docs.

**Comment by santos22** [comment]
Can we re-use the `bokeh-tree` role to link to the source file automatically?
```rst
PolyEditTool
~~~~~~~~~~~~~~

* name: ``'poly_edit'``
* menu icon: |poly_edit_icon|
* examples: :bokeh-tree:`sphinx/source/docs/user_guide/examples/tools_poly_draw.py`
```
